src/cart/crt_swim.c

```diff
@@ -445,30 +519,42 @@ int crt_swim_disable(struct crt_grp_priv *grp_priv, int crt_ctx_idx)
 
 	if (crt_ctx_idx < 0) {
 		swim_self_set(csm->csm_ctx, SWIM_ID_INVALID);
-		return 0;
+		D_GOTO(out, rc = 0);
 	}
 
 	D_RWLOCK_WRLOCK(&csm->csm_rwlock);
 	if (csm->csm_crt_ctx_idx == crt_ctx_idx) {
 		rc = crt_unregister_progress_cb(crt_swim_progress_cb,
-						csm->csm_crt_ctx_idx,
-						csm->csm_ctx);
+						csm->csm_crt_ctx_idx, NULL);
 		if (rc)
-			D_DEBUG(DB_TRACE, "crt_unregister_progress_cb() "
-				"failed: rc=%d\n", rc);
+			D_DEBUG(DB_TRACE, "crt_unregister_progress_cb()"
```

Comment:
same here, probably should be error instead

Reply:
done